test-case/check-runtime-pm-double-active.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+
+##
+## Case Name: check-runtime-pm-status
+## Preconditions:
+##    N/A
+## Description:
+##    check the audio runtime pm status
+## Case step:
+##    1. start aplay/arecord
+##    2. stop aplay/arecord
+##    3. sleep for runtime pm transition time
+##    4. check the runtime pm status
+## Expect result:
+##    command line check with $? without error
+##    runtime pm status must be suspended
+##    no error in dmesg
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+OPT_OPT_lst['t']='tplg'     OPT_DESC_lst['t']='tplg file, default value is env TPLG: $TPLG'
+OPT_PARM_lst['t']=1         OPT_VALUE_lst['t']="$TPLG"
+
+OPT_OPT_lst['l']='loop'     OPT_DESC_lst['l']='loop count'
+OPT_PARM_lst['l']=1         OPT_VALUE_lst['l']=3
+
+OPT_OPT_lst['d']='delay'    OPT_DESC_lst['d']='max delay time for state convert'
+OPT_PARM_lst['d']=1         OPT_VALUE_lst['d']=15
+
+OPT_OPT_lst['s']='sof-logger'   OPT_DESC_lst['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_PARM_lst['s']=0             OPT_VALUE_lst['s']=1
+
+# param: $1 -> max delay time for dsp pm status switch, unit is second
+func_check_dsp_status()
+{
+    # more frequent checking of the runtime pm status is required. Every loop is 100ms
+    local iloop=$(expr 10 '*' "$1")
+    dlogi "wait dsp power status to become suspended, starting max [100ms x $iloop times] checking"
+    for i in $(seq 1 $iloop)
+    do
+        # Here we pass a hardcoded 0 to python script, and need to ensure
+        # DSP is the first audio pci device in 'lspci', this is true unless
+        # we have a third-party pci sound card installed.
+        [[ $(sof-dump-status.py --dsp_status 0) == "suspended" ]] && break
+        sleep 0.1
+        if [ $i -eq $iloop ]; then
+            dlogi "dsp is not suspended after $1s, end test"
+            exit 1
+        fi
+    done
+    dlogi "dsp suspended in $i try, ${i}00 msec"
```

Comment:
how about use `return` instead of `break`
then after loop, you can do the error process

Reply:
@Bin-QA Do you mean `exit` or `break`

Reply:
@Bin-QA once DSP status is suspended, `break ` the loop and measure the time elapsed is the point. Error handling is pretty much timeout only.